### PR TITLE
Fix connection error handling

### PR DESCRIPTION
**Pull request by milton0825**
# TL;DR
Fix error handling for postgres connection error.

## Type
 - [X] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Gorm in the latest version switched the postgres connection library (from `github.com/lib/pq` to `github.com/jackc/pgconn`) and breaks the connection error handling in datacatalog.

And that is also causing issue the error handling in migration process. In https://github.com/flyteorg/datacatalog/blob/master/cmd/entrypoints/migrate.go#L42, the migration process should create the db if it does not exist but the error handling issue is causing it not to do so:
```
{"json":{"src":"migrate.go:67"},"level":"error","msg":"Failed to connect DB err failed to connect to `host=postgres user=postgres database=datacatalog`: server error (FATAL: database \"datacatalog\" does not exist (SQLSTATE 3D000))","ts":"2021-06-04T16:38:43Z"}
panic: failed to connect to `host=postgres user=postgres database=datacatalog`: server error (FATAL: database "datacatalog" does not exist (SQLSTATE 3D000))
goroutine 1 [running]:
github.com/flyteorg/datacatalog/cmd/entrypoints.glob..func1(0x2222a20, 0xc00021e820, 0x0, 0x2)
    /go/src/github.com/lyft/datacatalog/cmd/entrypoints/migrate.go:68 +0x987
github.com/spf13/cobra.(*Command).execute(0x2222a20, 0xc00021e800, 0x2, 0x2, 0x2222a20, 0xc00021e800)
    /go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:854 +0x2aa
github.com/spf13/cobra.(*Command).ExecuteC(0x2222cc0, 0x14d1c6f, 0x1, 0xffffffffffffffff)
    /go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:958 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
    /go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:895
github.com/flyteorg/datacatalog/cmd/entrypoints.Execute(0xc000000002, 0x405700)
    /go/src/github.com/lyft/datacatalog/cmd/entrypoints/root.go:44 +0x31
main.main()
    /go/src/github.com/lyft/datacatalog/cmd/main.go:10 +0x59
```

I did not catch this issue earlier in the sandbox testing because before I switched the datacatalog version, the migration process already creates the db hence I did not run into db not exist error. This time, I tried to re-create the postgres server before I run the latest datacatalog version and I can verify that the migration process succeeds. 

## Tracking Issue
NA

## Follow-up issue
NA

### CODE REVIEW

**Comment by milton0825** [comment]
cc: @katrogan  @kumare3

File: cmd/entrypoints/migrate.go [hunk 1]
```diff
@@ -38,7 +40,8 @@ var migrateCmd = &cobra.Command{
 
 		if err != nil {
 			// if db does not exist, try creating it
-			pqError, ok := err.(*pq.Error)
+			cErr, ok := err.(errors2.ConnectError)
+			pqError := cErr.Unwrap().(*pgconn.PgError)
```
Comment (katrogan) [inline cmd/entrypoints/migrate.go:44]: is this safe to do if !ok?

Comment (milton0825) [reply cmd/entrypoints/migrate.go:44]: wait. should have give it a second glance. Let me fix those

Comment (milton0825) [reply cmd/entrypoints/migrate.go:51]: fixed.

**Review by katrogan — APPROVED** [review]

**Comment by milton0825** [comment]
fixed DCO

**Review by katrogan — APPROVED** [review]

**Comment by milton0825** [comment]
Shall we merge it?
